modules/server/src/topics/profiletopic.cpp

```diff
@@ -36,13 +37,25 @@ bool ProfileTopic::isDone() const {
 }
 
 void ProfileTopic::handleJson(const nlohmann::json&) {
-    const std::string name = global::profile->meta->name.value_or("");
-
-    const nlohmann::json data = {
+    // @TODO (2025-04-30, emmbr) If we expose the json converters from profile.cpp, we
+    // could use those here instead and minimize the risk of getting the serialization of
+    // the data out of sync
+    nlohmann::json data = {
         { "uiPanelVisibility", global::profile->uiPanelVisibility },
         { "markNodes", global::profile->markNodes },
-        { "name", name }
+        { "filePath", global::configuration->profile}
     };
+
+    if (global::profile->meta.has_value()) {
+        data["name"] = global::profile->meta->name.value_or("");
```

Comment:
Does it make it easier on the Typescript side if the value always exist but might be empty?
Compared to:

```cpp
if (global::profile->meta->name.has_value()) {
    data["name"] = *global::profile->meta->name;
}
```

etc

Reply:
For the typescript side, it doesn't really matter. But since these are very "string-y" values, I personally think it could be nice if they are initialized with empty string, to avoid having to keep track of whether a license was set or not. 

However, what I would like us to do is to move all the JSON serialization code to one place, so that we could use the same serialization for the web UI as for the saving of the profile, for example. Hence the TODO-cmment at the top of this function. Now we have JSON serialization of the same thing in a few places, with a slightly different formatting. 

But that felt like too much work for this PR

Reply:
The drawback of that would be that if we change the format of the profile, the UI code would also have to change, however.